docs/threat-model.md

```diff
@@ -0,0 +1,553 @@
+---
+Authors: Sebastian Nagel
+Document Version: 1.0
+Last Updated: 2025-07-10
+Next Review: 2025-09-10
+---
+# Leios Consensus Upgrade - Threat Model
+
+A threat model for the Leios consensus change for Cardano. This was created by roughly following the [OWASP threat modeling process](https://owasp.org/www-community/Threat_Modeling_Process).
+
+## 1. System Overview
+
+> [!NOTE]
+> The described system here is the heavily simplified EB-only variant of Leios. Whenever we update this, reflect on existing assets, threats and mitigations, as well as add new ones accordingly.
+
+### 1.1 Description
+Leios is an overlay protocol on top of Ouroboros Praos that enhances transaction throughput by introducing Endorser Blocks (EBs) alongside regular Praos blocks (Ranking Blocks - RBs). The system maintains backward compatibility at the client interface while introducing new responsibilities for stake pools.
+
+### 1.2 Key Components
+
+#### Core Components
+- **Endorser Blocks (EBs)**: New block type that references transactions for inclusion, up to 640 kB (~20k transaction references)
+- **Ranking Blocks (RBs)**: Standard Praos blocks enhanced with EB certificates
+- **Vote Certificates**: Aggregated stake-weighted votes on EB validity and transaction availability
```

Comment:
In the first line these are called "EB certificates" but in the second line they are called "vote certificates". Perhaps we should use a single name for this component.